### PR TITLE
start of support С++20 

### PR DESCRIPTION
**Pull request by schnek**
**Description**: now it's not working properly...

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.

### CODE REVIEW

**Comment by schnek** [comment]
small fix for c++20 support it's finished rt 😅

File: src/world/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,9 @@
 # set up our project name
 project(world CXX)
 
-include(Game${ASCEMU_VERSION}/CMakeLists.txt)
+if ("${ASCEMU_VERSION}" STREQUAL "Mop" OR "${ASCEMU_VERSION}" STREQUAL "Cata" OR "${ASCEMU_VERSION}" STREQUAL "WotLK" OR "${ASCEMU_VERSION}" STREQUAL "TBC" OR "${ASCEMU_VERSION}" STREQUAL "Classic")
```
Comment (Zyres) [inline src/world/CMakeLists.txt:6]: There is always a ASCEMU_VERSION, why is it now a if statement?

**Comment by schnek** [comment]
> Check out your file changes, there are only changes in 3 files?

yes, 3 files. are still checked by CrashHandler and the set of functions (config) in this pr is not included due to a crash when start

> There is always a ASCEMU_VERSION, why is it now a if statement?

when building cmake, for some reason I sometimes see how
`include(Game/CMakeLists.txt)`

**Comment by schnek** [comment]
these changes are necessary, but cause a crash at startup, maybe you need to do something differently

https://github.com/AscEmu/AscEmu/commit/2f4503dde2d55d0be86d0bb01e02622c66b944cd#diff-c0bc0a4c888d88abe9db39f666bae0cb3c82441ec3229a189bfc5cbd5f3e5119R149

https://github.com/AscEmu/AscEmu/commit/2f4503dde2d55d0be86d0bb01e02622c66b944cd#diff-6d99d04a158109e2ba60a2f536031151574d46b7ae3a85d06a35bc3f9e34cd68R1887

https://github.com/AscEmu/AscEmu/commit/2f4503dde2d55d0be86d0bb01e02622c66b944cd#diff-1dce9af153e6a658e37a2e1cf6a71968f2860821683717f8c73502a2e68e1e47R254

**Comment by schnek** [comment]
fix  (https://github.com/AscEmu/AscEmu/commit/d26114a77e20ea64b1ff6b1c4cb0fd56b75482ff#)

**Pull request closed without merging**